questionnaire_linking.py

```diff
@@ -0,0 +1,136 @@
+    global QID_ERROR_COUNT
+    for line_number, questionnaire_link in enumerate(qid_link_file_reader, 1):
+        CASE_REF_ERROR_COUNT = 0
+        QID_ERROR_COUNT = 0
+        for i in questionnaire_link:
+            questionnaire_link[i] = questionnaire_link[i].replace(" ", "")
+        validate_qid_len_and_type(line_number, questionnaire_link["qid"])
+        validate_qid_third_digit(line_number, questionnaire_link["qid"])
+        validate_check_digits(line_number, questionnaire_link["qid"])
+        case_id = validate_case_ref(line_number, questionnaire_link["case_ref"])
+
+        if CASE_REF_ERROR_COUNT or QID_ERROR_COUNT:
+            print(f'Error: Line {line_number}: {questionnaire_link} has FAILED')
+        else:
+            post_message_to_queue(case_id, questionnaire_link["qid"], line_number)
+        print("\n")
+
+
+def validate_case_ref(line_number, case_ref):
+    global CASE_REF_ERROR_COUNT
+    if len(case_ref) != 8:
+        print(f'Error: Line {line_number}: Incorrect length of Case Reference {case_ref}. Expected length is 8 digits')
+        CASE_REF_ERROR_COUNT += 1
+    if not case_ref.isdigit():
+        print(f'Error: Line {line_number}: Incorrect type of Case Reference {case_ref}. Case Reference MUST be numeric')
+        CASE_REF_ERROR_COUNT += 1
+    if CASE_REF_ERROR_COUNT == 0:
+        response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/cases/ref/{case_ref}')
+        if response.status_code == 404:
+            print(f'Error: Line {line_number}: Case Reference {case_ref} does not exist')
+            CASE_REF_ERROR_COUNT += 1
+            return None
+        response.raise_for_status()
+        return response.json()["id"]
+
+
+def validate_qid_len_and_type(line_number, qid):
+    global QID_ERROR_COUNT
+    if len(qid) != 16:
+        print(f'Error: Line {line_number}: Incorrect length of QID {qid}. Expected length is 16 digits')
+        QID_ERROR_COUNT += 1
+    if not qid.isdigit():
+        print(f'Error: Line {line_number}: Incorrect type of QID {qid}. QID MUST be numeric')
+        QID_ERROR_COUNT += 1
+
+
+def validate_qid_third_digit(line_number, qid):
+    global QID_ERROR_COUNT
+    if len(qid) >= 3:
+        if qid[2] != "2":
+            print(f'Error: Line {line_number}: QID {qid} tranche identifier must be 2')
+            QID_ERROR_COUNT += 1
+    elif len(qid) < 3:
+        print(f'Error: Line {line_number}: QID {qid} must have a minimum of 3 digits for Tranche ID to be checked')
+
+
+def validate_check_digits(line_number, qid):
+    global QID_ERROR_COUNT
+    if len(qid) >= 3:
+        if not qid_checksum_validator.validate(qid, int(Config.QID_MODULUS), int(Config.QID_FACTOR))[0]:
+            print(f'Error: Line {line_number}: Check Digits incorrect in QID {qid}')
+            QID_ERROR_COUNT += 1
+    elif len(qid) < 3:
+        print(f'Error: Line {line_number}: QID {qid} length is not long enough for Check Digits to be validated')
```

Comment:
Should be incrementing the `QID_ERROR_COUNT` here?

Reply:
Sorted. I also missed this out on the function above this too.